test/KoreBuild.Tasks.Tests/VSWhereTests.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Runtime.InteropServices;
+using BuildTools.Tasks.Tests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace KoreBuild.Tasks.Tests
+{
+    public class VSWhereTests
+    {
+        private readonly ITestOutputHelper _output;
+
+        public VSWhereTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact]
+        public void ItFindsVisualStudio()
+        {
+            var engine = new MockEngine(_output);
+            var task = new VSWhere
+            {
+                BuildEngine = engine,
+            };
+
+            var result = task.Execute();
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                if (!result)
+                {
+                    // Don't fail the test. aspnet/BuildTools might be building without a version of VS installed.
+                    // We mostly want to make sure this task doesn't throw exceptions and fails gracefully
+                    Assert.NotEmpty(engine.Errors);
+                }
+                else
+                {
+                    Assert.NotEmpty(task.InstallationBasePath);
+                    Assert.True(Directory.Exists(task.InstallationBasePath), "VS installation was not found");
```

Comment:
Is it guaranteed that the that the installation base path will not be null? I'd assume so because we are requiring the nuget package.

Reply:
The installation base path should be non-null if the task succeeds. The task can fail if VS is not installed. vswhere is a standaone console tool that can be installed without installing vs.